src/save.ts

```diff
@@ -64,7 +65,15 @@ async function run() {
     }
 
     core.info(`... Saving cache ...`);
-    await cache.saveCache(config.cachePaths, config.cacheKey);
+    await withRetries(
+      () =>
+        withTimeout(
+          () => cache.saveCache(config.cachePaths, config.cacheKey),
+          config.timeout
+        ),
+      config.maxRetryAttempts,
+      () => true
+    );
```

Comment:
In my experience, saving the cache never failed. Caches are also a lot less often saved compared to being restored so hitting rate limits / getting bandwidth-limited is likely a lot less frequent.

I do understand though that we might want to keep this for consistencies' sake.